cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -360,40 +364,46 @@ static void sc_probe_nvidia_driver(struct sc_nvidia_driver *driver)
 			debug("nvidia driver version file doesn't exist");
 			driver->major_version = 0;
 			driver->minor_version = 0;
+			driver->micro_version = 0;
+			driver->version_string[0] = 0;
 			return;
 		}
 		die("cannot open file describing nvidia driver version");
 	}
-	// Driver version format is MAJOR.MINOR where both MAJOR and MINOR are
+	char *vres = fgets(driver->version_string, SC_NVIDIA_VERSION_MAX_LEN, file);
```

Comment:
I think just going with two-try call to `fscanf`would be good enough. You might need to encode the variant of the version string somehow, just in case we have to differentiate x.y.z with z == 0 from x.y. Perhaps a negative number could indicate that there's no micro component encoded.